examples/src/custom_frames/fib1/air.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use super::{BaseElement, FibEvaluationFrame, FieldElement, ProofOptions, TRACE_WIDTH};
+use crate::utils::are_equal;
+use winterfell::{
+    Air, AirContext, Assertion, EvaluationFrame, TraceInfo, TransitionConstraintDegree,
+};
+
+// FIBONACCI AIR
+// ================================================================================================
+
+pub struct FibAir {
+    context: AirContext<BaseElement>,
+    result: BaseElement,
+}
+
+impl Air for FibAir {
+    type BaseField = BaseElement;
+    type PublicInputs = BaseElement;
+    type Frame<E: FieldElement> = FibEvaluationFrame<E>;
+    type AuxFrame<E: FieldElement> = FibEvaluationFrame<E>;
+
+    // CONSTRUCTOR
+    // --------------------------------------------------------------------------------------------
+    fn new(trace_info: TraceInfo, pub_inputs: Self::BaseField, options: ProofOptions) -> Self {
+        let degrees = vec![TransitionConstraintDegree::new(1)];
+        assert_eq!(TRACE_WIDTH, trace_info.width());
+        let context =
+            AirContext::new(trace_info, degrees, 3, options).set_num_transition_exemptions(2);
+        FibAir {
+            context,
+            result: pub_inputs,
+        }
+    }
+
+    fn context(&self) -> &AirContext<Self::BaseField> {
+        &self.context
+    }
+
+    fn evaluate_transition<E: FieldElement + From<Self::BaseField>>(
+        &self,
+        frame: &Self::Frame<E>,
+        _periodic_values: &[E],
+        result: &mut [E],
+    ) {
+        let row_0 = frame.row(0);
```

Comment:
What about adding methods `current` and  `next` for the  `DefaultEvaluationFrame`? I think in that way you can keep the examples and simple Air programs almost unchanged.